routers/user/oauth_userinfo.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package user
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/modules/context"
+)
+
+// UserInfoResponse represents a successful userinfo response
+type UserInfoResponse struct {
+	Sub      string `json:"sub"`
+	Name     string `json:"name"`
+	Username string `json:"preferred_username"`
+	Email    string `json:"email"`
+	Picture  string `json:"picture"`
+}
+
+// UserInfoOAauth responds with OAuth formatted userinfo
+func UserInfoOAuth(ctx *context.Context) {
+	user := ctx.User
+	response := &UserInfoResponse{
+		Sub:      fmt.Sprint(user.ID),
+		Name:     user.FullName,
+		Username: user.Name,
+		Email:    user.Email,
```

Comment:
this should use the email function that respect privacy settings

Reply:
should it? Can't the user access their own email?

Reply:
uh yes - ctx.User is used ...